pattern-library/sass/patterns/_forms.scss

```diff
@@ -260,12 +260,15 @@ $input-width: (
 
     // Input fields immediately followed by a button (ie, search boxes) should eliminate the spacing
     // between them and adopt matching styling/sizing to emphasize that they are connected.
-    &[type=search] + button {
+    + button {
         border-top-left-radius: 0;
         border-bottom-left-radius: 0;
         font-size: font-size(base);
         margin-left: (-1 * $baseline / 4);
     }
+    &::-webkit-search-cancel-button {
+        display: none;
```

Comment:
Yes, I would remove lines 269-271 here and fix this issue in _normalize.scss (as it is a browser inconsistency problem, not a forms problem). You could do it by just directly reverting the changes I linked above.